### PR TITLE
[FLINK-22706][release]Update License information in NOTICE file

### PR DESCRIPTION
**Pull request by FuyaoLi2017**
## What is the purpose of the change

Flink documentation is migrated from Jekyll to Hugo through the PR [1], however, relevant LICENSE information is not updated in NOTICE file. This PR updates the dependencies to reflect the latest information.

I also updated the fontawesome version to reflect the current version being used in Flink.

In addition, bootstrap version 3.3.4 is known to have security vulnerabilities. See link [2].

I am not 100% sure about this change. I have a few questions and I am not sure about my assumptions below.

-  I didn't list the version of the hugo-book since it relies on Hugo and the version should be >=0.68 there [3].  
- The `docs/themes/book` is linking to an external github repository with a commitID. The actual path you see on Github is `docs/themes/book @ a486adf` I didn't specify the commitID in the description here since it could also be misleading. When you pull the flink repo. This will be intepreted as `docs/themes/book` path.
- There are some fourth party dependencies for hugo book. That includes KaTeX, Mermaid [4]. In [1], such license information is also mentioned in the commitID [5] but not merged into the Flink repo. As a result, I guess it is not necessary to list these fourth party libraries in this NOTICE, right? 

Thanks.
cc @rmetzger @sjwiesman @wuchong 

[1] https://github.com/apache/flink/pull/14903
[2] https://snyk.io/test/npm/bootstrap/3.3.4
[3] https://github.com/alex-shpak/hugo-book/blob/a486adf8462c0abfc9034436ddd72927d6656809/README.md#requirements
[4] https://github.com/alex-shpak/hugo-book/tree/a486adf8462c0abfc9034436ddd72927d6656809#shortcodes
[5] https://github.com/apache/flink/commit/bfe9602c77c7ed1729335006194c97b6ddea3900

### CODE REVIEW

**Comment by sjwiesman** [comment]
We do not include the Hugo git sub module as part of our source release. Additionally we do not release any build artifacts from Hugo. My understanding is that this means we do not need to include the transitive dependencies you find in themes/book

**Comment by zentol** [comment]
> we do not need to include the transitive dependencies you find in themes/book

This is correct.

File: NOTICE [hunk 1]
```diff
@@ -28,8 +22,9 @@ See bundled license files for details.
 This project bundles the following dependencies under SIL OFL 1.1 license (https://opensource.org/licenses/OFL-1.1).
 See bundled license files for details.
 
-- font-awesome:4.5.0 (Font) (http://fortawesome.github.io/Font-Awesome/) - Created by Dave Gandy
-    -> fonts in "docs/page/font-awesome/fonts"
+- font-awesome:4.6.3 (Font) (https://fontawesome.com/) - Created by Dave Gandy
```
Comment (zentol) [inline NOTICE:25]: How did you arrive at this version?

File: NOTICE [hunk 2]
```diff
@@ -7,17 +7,11 @@ The Apache Software Foundation (http://www.apache.org/).
...
-- Bootstrap v3.3.4 (http://getbootstrap.com) Copyright 2011-2015 Twitter, Inc.
-    -> css in "docs/page/css/bootstrap.min.css"
-    -> js in "docs/page/js/bootstrap.min.js"
-    -> fonts in "docs/page/fonts"
+    -> in "docs/static/js/anchor.min.js"
 
-Note: The "docs/page/fonts/glyphicons-halflings*" fonts are included in Bootstrap, and licensed under the same license as Boostrap.
-See https://glyphicons.com/old/license.html#old-halflings-bootstrap for more information.
+- hugo-book (https://github.com/alex-shpak/hugo-book) Copyright (c) 2018 Alex Shpak
+    -> in "docs/themes/book"
```
Comment (zentol) [inline NOTICE:15]:
```suggestion
```
This is not included in the source release.

File: NOTICE [hunk 3]
```diff
@@ -28,8 +22,9 @@ See bundled license files for details.
 This project bundles the following dependencies under SIL OFL 1.1 license (https://opensource.org/licenses/OFL-1.1).
 See bundled license files for details.
 
-- font-awesome:4.5.0 (Font) (http://fortawesome.github.io/Font-Awesome/) - Created by Dave Gandy
-    -> fonts in "docs/page/font-awesome/fonts"
+- font-awesome:4.6.3 (Font) (https://fontawesome.com/) - Created by Dave Gandy
+    -> css in "docs/static/font-awesome/css"
```
Comment (zentol) [inline NOTICE:26]: css files are licensed under MIT

**Review by zentol — CHANGES_REQUESTED** [review]
@sjwiesman The licenses directory contains a file for chroma, but this isn't referenced in the NOTICE. What's up with that?

File: NOTICE (hunk 1 above)
Comment (FuyaoLi2017) [reply NOTICE:25]: Hello Chesnay, I found the css version number to be 4.6.3 here.
https://github.com/apache/flink/blob/master/docs/static/font-awesome/css/font-awesome.css#L2
I am not sure if there is a version for fonts.

Comment (zentol) [reply NOTICE:25]: lets assume that the version applies to both

File: NOTICE (hunk 2 above)
Comment (FuyaoLi2017) [reply NOTICE:15]: Got it. I can delete these two lines. 
> I saw jquery and bootstrap were also only used and referenced in Flink doc so I intended to put hugo-book here (I will correct this).

File: NOTICE (hunk 3 above)
Comment (FuyaoLi2017) [reply NOTICE:26]: Thanks for pointing out. I will update the css file in the MIT part. I think the version of css is 4.6.3. Is there a version for the fonts?

**Comment by FuyaoLi2017** [comment]
> @sjwiesman The licenses directory contains a file for chroma, but this isn't referenced in the NOTICE. What's up with that?

This seems to be related to HUGO.
I found a explanation README here (https://github.com/gohugoio/hugo/blob/master/docs/content/en/commands/hugo_gen_chromastyles.md)

Maybe this is also considered as transitive dependencies? 
This indeed exists in the flink code as a css file. (https://github.com/apache/flink/blob/master/docs/assets/github.css)

If this need to be added to this notice file. Please suggest me the statement.

**Comment by sjwiesman** [comment]
@zentol because the syntax highlighting comes from a file generated by Hugo but which is actually chroma under the hood

The file is only called GitHub css because that’s the syntax theme

https://github.com/apache/flink/blob/master/docs/assets/github.css

**Comment by FuyaoLi2017** [comment]
> @zentol because the syntax highlighting comes from a file generated by Hugo but which is actually chroma under the hood
> 
> The file is only called GitHub css because that’s the syntax theme
> 
> https://github.com/apache/flink/blob/master/docs/assets/github.css

@sjwiesman Do you think we need to add the chroma to the NOTICE file in MIT license part? Something like this? 

```
- chroma (css generated by Hugo) (https://github.com/alecthomas/chroma) Copyright (C) 2017 Alec Thomas
    -> in "docs/assets/github.css"
```

**Comment by sjwiesman** [comment]
Yes

**Comment by FuyaoLi2017** [comment]
cc @zentol @sjwiesman  Hello All, I think this PR should be ready. Please review.
In addition, I have a side question, there are many third party dependencies over the places of flink source code. For example, kubernetes part uses fabric8io library. (https://github.com/apache/flink/blob/master/flink-kubernetes/pom.xml#L73)

The NOTICE file doesn't contain these information. Could you explain a bit?

**Comment by zentol** [comment]
> The NOTICE file doesn't contain these information. Could you explain a bit?

The NOTICE file in the root of the project is only for the source release. It only covers files are that actually contained in that releases, i.e., the contents of the repository with various exceptions.
NOTICE files in generally only cover what is included in an artifact; flink-kubernetes contains a NOTICE file that covers all dependencies that we bundle in the flink-kubernetes jar; dependencies that aren't bundled do not need to be listed in the NOTICE.

**Pull request merged**